packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -186,7 +199,36 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     /**
      * {@inheritDoc (ISharedObject:interface).snapshot}
      */
-    public abstract snapshot(): ITree;
+    public summarize(fullTree: boolean = false): IChannelSummarizeResult {
+        // Create a SummarySerializer that will be used to serialize any IFluidHandles in this object. The
+        // IFluidHandles represents route to any referenced Fluid object.
+        // SummarySerializer tracks the routes of all handles that it serializes.
+        const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
+
+        const snapshot: ITree = this.snapshotCore(serializer);
+        const summaryTree = convertToSummaryTreeWithStats(snapshot, fullTree);
+
+        const node: IGarbageCollectionNode = {
+            path: this.id,
```

Comment:
Historically, we had problems in routing because we allowed multiple path formatting and did not normalize paths in right places. Do we normalize paths as we process them? I.e. I'd assume "id" & "/id" are the same thing. Or do we only support (and enforce that) paths without leading slash? What about trailing slash?
Please either enforce across all layers specific format, or normalize paths across all layers that work with pahts.

Reply:
I will make sure we have a uniform format for this across all layers. I have removed this from here and will re-add in a separate PR.